recipes/cabinetry/meta.yaml

```
@@ -0,0 +1,60 @@
+{% set name = "cabinetry" %}
+{% set version = "0.4.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 88accfd0a6138e1b0c7ce0b5f44a746a53ebed8488bea02a5333846fee71f30a
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  entry_points:
+    - cabinetry = cabinetry.cli:cabinetry
+
+requirements:
+  host:
+    - python >=3.7
+    - pip
+  run:
+    - python >=3.7
+    - pyhf[minuit] ~=0.6.3
+    - iminuit
```

Maybe worth putting a bound on `iminuit` that matches `pyhf`'s. c.f. https://github.com/scikit-hep/pyhf/blob/8e242c9c829963c714f9b10acd28494f004a2452/setup.py#L12

```suggestion
    - iminuit >=2.7.0
```

---

Are we not picking up this bound indirectly through `pyhf` anyway? It is not clear to me why `iminuit` needs to be listed but some other direct dependencies that are picked up through `pyhf` (like `pyyaml`) do not.

---

`pyyaml` is used here: https://github.com/scikit-hep/cabinetry/blob/00ffff5be9a03c3d56b0a4c54749045bc5a2cecd/src/cabinetry/configuration.py#L10. Is this perhaps related to `iminuit` being a setup extra for `pyhf` will `pyyaml` is a core dependency for `pyhf` (see https://github.com/conda-forge/pyhf-feedstock/blob/654fd1612692d958e193ed2fd43a8ada92a9943f/recipe/meta.yaml#L32)?

---

Not sure. I agree it seems strange that we need to explicitly place it here though. I unfortunately need to run so I can't offer any advice right now &mdash; sorry!

---

> Maybe worth putting a bound on iminuit that matches `pyhf`

This should probably be a `run-constrained` requirement in the `pyhf` recipe (i.e. "isn't a required dependency but if it's installed it should meet this requirement).